### ISSUE TITLE
Building at user-devnet tag is impossible because fil-proofs repo was renamed

### ISSUE THREAD

**Issue by sywyn219** [filecoin-project/go-filecoin#2211]
### Description

16:21:10.061 ERROR       node: processBlock(): processing block from network: input chain co
How can I solve this error?

I executed rm -rf ~/.filecoin and tried it again many times, always reporting this error. Regardless of how long the sync is, the block height is always 0

**Comment by rkowalick** [comment]
@sywyn219

If you are connecting to the usernet, you need to use the correct code from the instructions in https://github.com/filecoin-project/go-filecoin/wiki/Getting-Started.

Did you run
```
git fetch origin devnet-user -f
git checkout devnet-user
```
before running `deps` and `build`? The user devnet won't work with the latest build.

**Comment by sywyn219** [comment]
If I executed:
Git fetch origin devnet-user -f
Git checkout devnet-user

Will happen:
Before
Root@iZj6chddomqhdb241yn4qsZ:~/go/code/src/github.com/filecoin-project/go-filecoin/scripts# ls
Install-rust-fil-proofs.sh

After
This file becomes:
Install-rust-proofs.sh

This error will be released at compile time.
Using precompiled rust-proofs
$GITHUB_TOKEN not set
Failed to find or obtain precompiled rust-proofs, falling back to local
Cloning into '/root/go/code/src/github.com/filecoin-project/go-filecoin/bls-signatures/bls-signatures'...
Git@github.com: Permission denied (publickey).
Fatal: Could not read from remote repository.

**Comment by dayou5168** [comment]
@sywyn219 
i got the same issue！

**Comment by sywyn219** [comment]
There were some problems at my last description , I mean that 
Before I executed ：

git fetch origin devnet-user -f
git checkout devnet-user

it happens
Root@iZj6chddomqhdb241yn4qsZ:~/go/code/src/github.com/filecoin-project/go-filecoin/scripts# ls
Install-rust-fil-proofs.sh

After i executed that
it became like this 
Install-rust-proofs.sh

and then when i compile it goes error

Using precompiled rust-proofs
$GITHUB_TOKEN not set
Failed to find or obtain precompiled rust-proofs, falling back to local
Cloning into '/root/go/code/src/github.com/filecoin-project/go-filecoin/bls-signatures/bls-signatures'...
Git@github.com: Permission denied (publickey).
Fatal: Could not read from remote repository.

**Comment by pengxiankaikai** [comment]
@sywyn219 @rkowalick Hello! I also encountered the same problem when compiling with the devnet branch.

```
using precompiled rust-proofs
$GITHUB_TOKEN not set
failed to find or obtain precompiled rust-proofs, falling back to local
Cloning into '/home/karl/projects/go/src/github.com/filecoin-project/go-filecoin/bls-signatures/bls-signatures'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:filecoin-project/bls-signatures.git' into submodule path '/home/karl/projects/go/src/github.com/filecoin-project/go-filecoin/bls-signatures/bls-signatures' failed
Failed to clone 'bls-signatures/bls-signatures'. Retry scheduled
Cloning into '/home/karl/projects/go/src/github.com/filecoin-project/go-filecoin/proofs/rust-proofs'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:filecoin-project/rust-proofs.git' into submodule path '/home/karl/projects/go/src/github.com/filecoin-project/go-filecoin/proofs/rust-proofs' failed
Failed to clone 'proofs/rust-proofs'. Retry scheduled
Cloning into '/home/karl/projects/go/src/github.com/filecoin-project/go-filecoin/bls-signatures/bls-signatures'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:filecoin-project/bls-signatures.git' into submodule path '/home/karl/projects/go/src/github.com/filecoin-project/go-filecoin/bls-signatures/bls-signatures' failed
Failed to clone 'bls-signatures/bls-signatures' a second time, aborting
~/projects/go/src/github.com/filecoin-project/go-filecoin/proofs/rust-proofs ~/projects/go/src/github.com/filecoin-project/go-filecoin
cargo 1.32.0 (8610973aa 2019-01-02)
error: could not find `Cargo.toml` in `/home/karl/projects/go/src/github.com/filecoin-project/go-filecoin/proofs/rust-proofs` or any parent directory
error: could not find `Cargo.toml` in `/home/karl/projects/go/src/github.com/filecoin-project/go-filecoin/proofs/rust-proofs` or any parent directory
```

**Comment by anorth** [comment]
Ah, I see what has happened here.
- The `rust-proofs` repository was renamed to `rust-fil-proofs` after the `devnet-user` tag, but that name change is outside the scope of git history, so the build from source won't work.
- At `master`, the `GITHUB_TOKEN` env var is no longer required so we've [removed it from documentation](https://github.com/filecoin-project/go-filecoin/commit/65268f6a4f860f3324be1f2a4579a326e70ffa57). The prebuilt proofs blob is not yet officially supported: #1913

This is a tough spot. Building the `devnet-user` tag as is just not gunna work after the repo rename. cc @dignifiedquire @porcuquine .

One option would be to turn `devnet-user` into a branch where we can fix the build script for the rename.

**Comment by porcuquine** [comment]
If introducing a branch you can hotfix doesn't interfere with your release process requirements, that seems like a sensible solution. (And if it does interfere, maybe the process can be amended in light of this need, which might repeat itself in the future for any number of reasons.)

**Comment by anorth** [comment]
I believe our release process will move to branches in the future.

I'm scoping this issue to fixing the immediate impediment until our next release (2-3 weeks) and raising to P0 since it blocks anyone joining the network.

**Comment by porcuquine** [comment]
I'm not familiar enough with the release process, so this might not make sense. But if the requirement is just for 'a tag', rather than 'the tag', could you make the branch and tag with `devnet-user-fix` or whatever and document the change?

**Comment by mishmosh** [comment]
Thank you @gmasgras/@ognots for looking into this. If this takes longer than a day, please flag. 

For long-term usability, better to ensure we can deliver the good, reliable release process you are currently designing.

**Comment by phritz** [comment]
@ognots @gmas @porcuquine per https://docs.google.com/document/d/1p3YQF_YsFwyI6UYd4eNkf_OPQPXcD8gBDR8NTm86zJI/edit we should be using a branch for the next release, please ack. (sorry to community contributors, doc isn't open yet, we are working to move our docs into the open).

**Comment by gmasgras** [comment]
I've branched out from the `devnet-user` tag into `devnet-user-0.0.2` and updated the submodules to point at the (renamed) `rust-fil-proof`. Circle was able to build successfully (https://circleci.com/gh/filecoin-project/go-filecoin/12222) but it used the cached proof-params. It would've most likely timed out would it have needed to generate the params from scratch. 

Note that users would have to wait 1hr+ for this branch of go-filecoin to build (including generating params) because the version of `rust-fil-proofs` at this commit does not include the `paramfetch` binary that automatically downloads the params from IPFS at build time. I'll include a link to the params along with instructions on how to build using cached params. 

We could also create a `0.0.2` release on GitHub providing Linux/OSX binaries and proofs params, (it would look something like https://github.com/filecoin-project/go-filecoin/releases/tag/9.9.999) in order to make it easier for users to join.

**Comment by gmasgras** [comment]
@mishmosh @phritz @ognots 
I've updated the [Getting Started](https://github.com/filecoin-project/go-filecoin/wiki/Getting-Started#install-filecoin-and-its-dependencie) page with instructions on how to build go-filecoin from source (devnet-user tag) or use the binary from the [0.0.2 Github Release](https://github.com/filecoin-project/go-filecoin/releases/tag/0.0.2)

**Comment by anorth** [comment]
Great stuff, thank you so much @gmasgras.

@sywyn219 @dayou5168 @pengxiankaikai or anyone else, if you could try the new instructions and report back, we'd be very appreciative!

**Comment by anorth** [comment]
This was resolved properly in release 0.1
